In [1]:
import pickle 
import gzip
import numpy as np


In [8]:
f = gzip.open("/Users/harishprabhu/Documents/GitHub/Neural_Network_from_Scratch/data/mnist.pkl.gz", 'rb')
u = pickle._Unpickler(f)
u.encoding = 'latin1'
training_data, validation_data, test_data = u.load()


In [9]:
training_data

(array([[0., 0., 0., ..., 0., 0., 0.],
        [0., 0., 0., ..., 0., 0., 0.],
        [0., 0., 0., ..., 0., 0., 0.],
        ...,
        [0., 0., 0., ..., 0., 0., 0.],
        [0., 0., 0., ..., 0., 0., 0.],
        [0., 0., 0., ..., 0., 0., 0.]], dtype=float32),
 array([5, 0, 4, ..., 8, 4, 8]))

In [14]:
print(training_data[0].shape)
print(training_data[1].shape)

(50000, 784)
(50000,)


we are selelcting only the images that is (784,) and reshaping(flattening) to (784, 1) to feed in to the FULLY CONNECTED NEURAL NETWORKS.

In [11]:
training_data[0].shape

(50000, 784)

In [17]:
validation_data[0].shape

(10000, 784)

In [18]:
test_data[0].shape

(10000, 784)